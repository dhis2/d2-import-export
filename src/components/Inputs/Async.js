import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_YES = { value: 'true', label: i18n.t('Yes') }
export const OPTION_NO = { value: 'false', label: i18n.t('No') }
export const ASYNC_KEY = 'async'
export const ASYNC_DEFAULT_VALUE = OPTION_YES.value

export const Async = () => (
    <RadioGroup
        name={ASYNC_KEY}
        label={i18n.t('Async')}
        options={[OPTION_YES, OPTION_NO]}
    />
)
