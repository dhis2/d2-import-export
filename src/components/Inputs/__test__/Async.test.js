import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Async } from '../Async'

jest.mock('react-final-form')

describe('Input component - Async', () => {
    it('should render correctly', () => {
        const file = shallow(<Async />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
