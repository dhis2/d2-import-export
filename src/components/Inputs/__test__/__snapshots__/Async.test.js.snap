// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Async should render correctly 1`] = `
<RadioGroup
  label="Async"
  name="async"
  options={
    Array [
      Object {
        "label": "Yes",
        "value": "true",
      },
      Object {
        "label": "No",
        "value": "false",
      },
    ]
  }
/>
`;
